ansible/vm_files/playbook.yml

```diff
@@ -0,0 +1,61 @@
+- name: Test Playbook
+  hosts: webservers
+  remote_user: ec2-user
+
+  vars_prompt:
+    - name: trello_secret_key
+      private: yes
+      prompt: Enter your trello secret key
+
+    - name: trello_secret_token
+      private: yes
+      prompt: Enter your trello secret token
+
+  tasks:
+  - name: Install Dependencies
+    ansible.builtin.yum:
+      name:
+        - python3
+        - git
+    become: yes
+
+  - name: Install poetry
+    ansible.builtin.shell: curl -sSL https://install.python-poetry.org | python3 -
+    args:
+      creates: ~/.local/bin/poetry
+
+  - name: Create Directory for Todo app
+    ansible.builtin.file:
+      path: /opt/todoapp
+      state: directory
+      mode: "777"
+    become: yes
+  
+  - name: Clone git repository
+    ansible.builtin.git:
+      repo: https://github.com/SChandarana/Devops-Course-Starter
+      dest: /opt/todoapp
+      version: main
+  
+  - name: Install Project Dependencies
+    ansible.builtin.shell: source ~/.bash_profile; poetry install
+    args: 
+      chdir: /opt/todoapp
+
+  - name: Create .env file
+    ansible.builtin.template:
+      src: ~/vm_files/.env.j2
+      dest: /opt/todoapp/.env
+  
+  - name: Copy service file
+    ansible.builtin.copy:
+      src: /opt/todoapp/ansible/todoapp.service
+      dest: /etc/systemd/system
+      remote_src: yes
+    become: yes
```

Comment:
This won't work currently as the todoapp.service hasn't been pushed to main (I was using this git branch when testing it all worked but the final playbook uses the main branch when it clones git)

Reply:
Makes sense!